### ISSUE TITLE
Feature Request: Event Coordinates

### ISSUE THREAD

**Issue by pigulla** [unicorncoding/devevents-web#668]
Hey @unicorncoding,

as briefly discussed in #665 we're integrating your API into an internal application (more updates on that soon, hopefully) and one feature that we thought would be great is if there was location data for offline events.

While there are `countryCode`  and `cityCode` fields in the API, those are not sufficient to infer the actual location. For cities like "San Francisco" it might be feasible to guess, but in Germany "Frankfurt" is already a problem because there's "Frankfurt am Main" and "Frankfurt an der Oder".

My suggestion would be to add `latitude` and `longitude` to the database and ask people to include that information in PRs. It takes less than a minute to get these coordinates from Wikipedia. These coordinates could then be mapped back to an actual city using [Reverse geocoding](https://developers.google.com/maps/documentation/geocoding/requests-reverse-geocoding).

Adding this feature would enable all sorts of nice functionality, such as filtering events by distance from your location, finding events near each other or displaying a map of the world with all upcoming events.

The dev.events backend is no longer a public repository, but if you're interested in a helping hand, I'd be happy to whip up a PR!

**Comment by unicorncoding** [comment]
Hey @pigulla 

Will lack of lat/long at this time impede the rollout of the first version of your app or you're good enough with the current data? I may add lat/long in the future, but there are some challenges involved, including geo-encoding existing data which is pretty much consistent, but doesn't respect official city names (e.g. Frankfurt am Main is just Frankfrurt, Democratic republic of Congo is Congo, Málaga is Malaga – for brevity and SEO reasons).

**Comment by pigulla** [comment]
No, not at all. It's just a feature that would be cool but it is in no way a must-have.

Should that feature be added at some point, it would probably be optional anyway, at least for quite some time. I don't think retroactively adding those fields to all events in the database is practical.

I fully agree with your UX concerns. Maybe using an API like Geonames would be useful (e.g., [this](https://www.geonames.org/advanced-search.html?q=Frankfurt&country=&featureClass=P&continentCode=)) or maybe Google Maps.

Let me know if there's anything I could help with!

**Comment by unicorncoding** [comment]
Nice, thanks! Since the lat/lon could be useful in the feature, let's then keep this issue open and see where it goes from here.
